test/functional/eval/special_vars_spec.lua

```diff
@@ -168,4 +169,11 @@ describe('Special values', function()
       'Expected True but got v:null',
     }, meths.get_vvar('errors'))
   end)
+
+  describe('compat', function()
+    it('v:count is distinct from count', function()
+      command('let count = v:count + 1')
+      neq(eval('count'), eval('v:count'))
+    end)
+  end)
```

Comment:
I would not use `neq`, just set it to list and check that it works as you could not set `v:count` to list. Also good idea to check `eq(1, eval('count is# g:["count"]')`.

Reply:
Ok!